static/js/components/channels/withChannelCreateDialog.js

```diff
@@ -0,0 +1,123 @@
+// @flow
+/* global SETTINGS: false */
+import React from "react"
+import R from "ramda"
+
+import { showDialog, hideDialog, setToastMessage } from "../../actions/ui"
+import {
+  startChannelEdit,
+  clearChannelEdit,
+  updateChannelEdit,
+  createChannel
+} from "../../actions/channels"
+import { TOAST_SUCCESS } from "../../constants"
+import { actions } from "../../lib/redux_rest"
+import { discussionErrors } from "../../lib/validation/discussions"
+import { getDisplayName } from "../../util/util"
+import { CHANNEL_CREATE_DIALOG } from "../../constants"
+import ChannelCreateDialog from "./ChannelCreateDialog"
+
+import type { CreateChannelResponse } from "../../flow/discussionTypes"
+
+const isVisible = R.propOr(false, CHANNEL_CREATE_DIALOG)
+
+export const withChannelCreateDialog = (WrappedComponent: ReactClass<*>) => {
+  class WithChannelCreateDialog extends React.Component {
+    openChannelCreateDialog = (searchkit: Object): void => {
+      const { dispatch } = this.props
+      dispatch(
+        startChannelEdit({
+          filters: searchkit.query.getSelectedFilters(),
+          searchkit
+        })
+      )
+      dispatch(showDialog(CHANNEL_CREATE_DIALOG))
+    }
+
+    closeAndClearDialog = (): void => {
+      const { dispatch } = this.props
+      dispatch(clearChannelEdit())
+      dispatch(hideDialog(CHANNEL_CREATE_DIALOG))
+    }
+
+    closeAndCreateDialog = (): void => {
+      const { dispatch, channelDialog: { inputs, searchkit } } = this.props
+      const query = searchkit.buildQuery().query
+      new Promise((resolve, reject) => {
+        if (R.isEmpty(discussionErrors(inputs))) {
+          resolve()
+        } else {
+          reject()
+        }
+      }).then(() => {
+        dispatch(
+          createChannel((...args) => dispatch(actions.channels.post(...args)), [
+            { ...inputs, query }
+          ])
+        ).then((channel: CreateChannelResponse) => {
+          this.closeAndClearDialog()
+
+          if (SETTINGS.open_discussions_redirect_url) {
+            const channelUrl = `${SETTINGS.open_discussions_redirect_url}channel/${channel.name}`
+            window.open(channelUrl, "_blank")
```

Comment:
This wasn't working for me, and I found this which might explain it: https://stackoverflow.com/questions/11821009/javascript-window-open-not-working

Reply:
Yeah, I had to allow it in my popup blocker too. I don't think that's avoidable since we need to wait until the channel is created to open it.